External/FEXCore/Source/Interface/IR/Passes/RegisterAllocationPass.cpp

```diff
@@ -254,13 +255,18 @@ namespace {
   }
 
   void FreeRegisterGraph(RegisterGraph *Graph) {
+    if (Graph->Nodes) {
+      ::munmap(Graph->Nodes, Graph->NodeCount * sizeof(RegisterNode));
+    }
     delete Graph;
   }
 
   void ResetRegisterGraph(RegisterGraph *Graph, uint64_t NodeCount) {
     NodeCount = AlignUp(NodeCount, sizeof(uint64_t));
-    Graph->Nodes.clear();
-    Graph->Nodes.resize(NodeCount);
+    if (Graph->Nodes) {
+      ::munmap(Graph->Nodes, Graph->NodeCount * sizeof(RegisterNode));
```

Comment:
This is pretty hot code.

Wouldn't it be much faster to just memset the memory if the allocated space is already large enough?

Reply:
You're right. I thought the check for that case was elsewhere which is why I didn't handle it there.